service/k8sclient/k8client.go

```diff
@@ -378,3 +413,239 @@ func (c *K8Client) setXtraDBComputeResources(podResources *pxc.PodSpec, computeR
 		}
 	}
 }
+
+// ListPSMDBClusters returns list of psmdb clusters and their statuses.
+func (c *K8Client) ListPSMDBClusters(ctx context.Context) ([]PSMDBCluster, error) {
+	clusters, err := c.getPSMDBClusters(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	deletingClusters, err := c.getDeletingPSMDBClusters(ctx, clusters)
+	if err != nil {
+		return nil, err
+	}
+	res := append(clusters, deletingClusters...)
+
+	return res, nil
+}
+
+// CreatePSMDBCluster creates percona server for mongodb cluster with provided parameters.
+func (c *K8Client) CreatePSMDBCluster(ctx context.Context, params *PSMDBParams) error {
+	res := &perconaServerMongoDB{
+		TypeMeta: TypeMeta{
+			APIVersion: psmdbAPIVersion,
+			Kind:       string(perconaServerMongoDBKind),
+		},
+		ObjectMeta: ObjectMeta{
+			Name: params.Name,
+		},
+		Spec: perconaServerMongoDBSpec{
+			Image: psmdbImage,
+			Secrets: &secretsSpec{
+				Users: "my-cluster-name-secrets",
+			},
+			Mongod: &mongodSpec{
+				Net: &mongodSpecNet{
+					Port: 27017,
+				},
+				OperationProfiling: &mongodSpecOperationProfiling{
+					Mode:              operationProfilingModeSlowOp,
+					SlowOpThresholdMs: 100,
+					RateLimit:         1,
+				},
+				Security: &mongodSpecSecurity{
+					RedactClientLogData:  false,
+					EnableEncryption:     pointer.ToBool(true),
+					EncryptionKeySecret:  "my-cluster-name-mongodb-encryption-key",
+					EncryptionCipherMode: mongodChiperModeCBC,
+				},
+				SetParameter: &mongodSpecSetParameter{
+					TTLMonitorSleepSecs:                   60,
+					WiredTigerConcurrentReadTransactions:  128,
+					WiredTigerConcurrentWriteTransactions: 128,
+				},
+				Storage: &mongodSpecStorage{
+					Engine: storageEngineWiredTiger,
+					InMemory: &mongodSpecInMemory{
+						EngineConfig: &mongodSpecInMemoryEngineConfig{
+							InMemorySizeRatio: 0.9,
```

Comment:
My recommendation would be to remove all those optional things for now and let operator decide. It will be easier to add them later than to change them

Reply:
Ok, I just copied values from `cr.yaml`, decided that it's a recommended values.

Reply:
I asked about it – that's just examples, and defaults are not currently documented. Some fields have them, and some don't. We should keep only required for now

Reply:
ok

Reply:
That wasn't done, isn't it?